tests/chains/ethereum/scripts/cmd/helper/chain.go

```diff
@@ -0,0 +1,220 @@
+		IBCCommitment: *ibcCommitment,
+
+		SimpleToken:   *simpletoken,
+		ICS20Transfer: *ics20transfer,
+		ICS20Bank:     *ics20bank,
+	}
+}
+
+func (chain *Chain) TxOpts(ctx context.Context, index uint32) *bind.TransactOpts {
+	return makeGenTxOpts(big.NewInt(chain.chainID), chain.prvKey(index))(ctx)
+}
+
+func (chain *Chain) prvKey(index uint32) *ecdsa.PrivateKey {
+	key, ok := chain.keys[index]
+	if ok {
+		return key
+	}
+	key, err := wallet.GetPrvKeyFromMnemonicAndHDWPath(chain.mnemonicPhrase, fmt.Sprintf("m/44'/60'/0'/0/%v", index))
+	if err != nil {
+		panic(err)
+	}
+	chain.keys[index] = key
+	return key
+}
+
+func (chain *Chain) CallOpts(ctx context.Context, index uint32) *bind.CallOpts {
+	opts := chain.TxOpts(ctx, index)
+	return &bind.CallOpts{
+		From:    opts.From,
+		Context: opts.Context,
+	}
+}
+
+func (chain *Chain) LastHeader() *gethtypes.Header {
+	return chain.LastLCState.Header()
+}
+
+func (chain *Chain) UpdateHeader() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	for {
+		state, err := chain.lc.GetState(ctx, chain.ContractConfig.GetIBCHandlerAddress(), nil, nil)
+		if err != nil {
+			panic(err)
+		}
+		if chain.LastLCState == nil || state.Header().Number.Cmp(chain.LastHeader().Number) == 1 {
+			chain.LastLCState = state
+			return
+		} else {
+			continue
+		}
+	}
+}
+
+func makeGenTxOpts(chainID *big.Int, prv *ecdsa.PrivateKey) func(ctx context.Context) *bind.TransactOpts {
+	signer := gethtypes.LatestSignerForChainID(chainID)
+	addr := gethcrypto.PubkeyToAddress(prv.PublicKey)
+	return func(ctx context.Context) *bind.TransactOpts {
+		return &bind.TransactOpts{
+			From:     addr,
+			GasLimit: 6382056,
+			Signer: func(address common.Address, tx *gethtypes.Transaction) (*gethtypes.Transaction, error) {
+				if address != addr {
+					return nil, errors.New("not authorized to sign this account")
+				}
+				signature, err := gethcrypto.Sign(signer.Hash(tx).Bytes(), prv)
+				if err != nil {
+					return nil, err
+				}
+				return tx.WithSignature(signer, signature)
+			},
+		}
+	}
+}
+
+type ContractConfig interface {
+	GetIBCHandlerAddress() common.Address
+	GetIBCCommitmentTestHelperAddress() common.Address
+
+	GetSimpleTokenAddress() common.Address
+	GetICS20TransferBankAddress() common.Address
+	GetICS20BankAddress() common.Address
+}
+
+type Connection struct {
+	ID                   string
+	ClientID             string
+	CounterpartyClientID string
+	NextChannelVersion   string
+	Channels             []Channel
+}
+
+type Channel struct {
+	PortID               string
+	ID                   string
+	ClientID             string
+	CounterpartyClientID string
+	Version              string
+}
+
+func CreateChannel() Channel {
+	return Channel{
+		PortID:               "transfer",
+		ID:                   "channel-0",
+		ClientID:             "mock-client-0",
+		CounterpartyClientID: "mock-client-0",
+		Version:              "transfer-1",
+	}
+}
```

Comment:
You shouldn't hard-code test-specific constant values in these CLI tools.

Reply:
The definition here is.
https://github.com/dongrie/yui-relayer-build/blob/v0.3/tests/cases/eth2eth/configs/relayer/path.json
I think I need to match this, can I pass it as a CLI argument?

Reply:
I think there are two options.
1. Provide a path to `path.json` as a CLI argument, and then the tools will parse the json and get required values.
2. Provide each value (e.g. `src-chain-id`, `src-client-id`, ...) as a CLI argument.

Anyway, you need to provide values not found in existing config files to the CLI tools (e.g. `token-address`, `sender-address`, ...)

Reply:
Changed to pass path.json as a CLI argument.
https://github.com/datachainlab/yui-relayer-build/pull/5/commits/31878bebacedab0c409a4ab7623942c5f65d4803